### DISCUSSION TITLE
Can I use the soft-dtw loss of tslearn?

### DISCUSSION THREAD

**Discussion by weidongzhou1994** [ourownstory/neural_prophet#1410] [Q&A - get help using NeuralProphet]
As title, I want to use the soft-dtw loss, but when i use it, it give me this error: TypeError: forward() missing 2 required positional arguments: 'x' and 'y'

**Comment by leoniewgnr** [comment]
Hi @weidongzhou1994, sorry for the long wait. NeuralProphet supports all loss functions that are listed here: https://pytorch-enhance.readthedocs.io/en/latest/_modules/torch/nn/modules/loss.html
I think `soft-dtw` is not listed, so I'm afraid it is not possible or how did you try to use?

**Reply by weidongzhou1994** [reply]
I use the [soft-dtw](https://tslearn.readthedocs.io/en/stable/auto_examples/autodiff/plot_soft_dtw_loss_for_pytorch_nn.html#soft-dtw-loss-for-pytorch-neural-network) of tslearn. So except  loss functions that are listed in pytorch, can i write the loss function by myself.

**Reply by leoniewgnr** [reply]
Yes this should be possible: https://towardsdatascience.com/implementing-custom-loss-functions-in-pytorch-50739f9e0ee1
Let me know how it goes!:)
